Objective-C/CBLReplicator.mm

```diff
@@ -593,14 +593,16 @@ - (void) resolveConflict: (CBLReplicatedDocument*)doc {
 - (void) _resolveConflict: (CBLReplicatedDocument*)doc {
     CBLLogInfo(Sync, @"%@: Resolve conflicting version of '%@'", self, doc.id);
     NSError* error;
-    if ([_config.database resolveConflictInDocument: doc.id
-                               withConflictResolver: _config.conflictResolver
-                                              error: &error])
+    BOOL success = [_config.database resolveConflictInDocument: doc.id
+                                          withConflictResolver: _config.conflictResolver
+                                                         error: &error];
+    if (success && !error) {
```

Comment:
Is there a case of success and having an error?

Reply:
Can we do something this?

```
- (void) _resolveConflict: (CBLReplicatedDocument*)doc {
    CBLLogInfo(Sync, @"%@: Resolve conflicting version of '%@'", self, doc.id);
    NSError* error = nil;
    BOOL success = [_config.database resolveConflictInDocument: doc.id
                                          withConflictResolver: _config.conflictResolver
                                                         error: &error];
    if (!success)
        CBLWarn(Sync, @"%@: Conflict resolution of '%@' failed: %@", self, doc.id, error);
    
    [doc updateError: error];
    [self logErrorOnDocument: doc pushing: NO];
    [self postDocumentReplications: @[doc] pushing: NO];
}
```

Reply:
i think, if we return true from the database, this is enough.

Reply:
> Is there a case of success and having an error?
there is none i can think of. i think, only checking success is fine

Reply:
Then in updateError, if error is nil, you can internally resetError so don't need two methods, resetError and updateError.